clusterloader/framework/pod_utils.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const maxRetries = 5
+
+// CreatePods creates pods in a user defined namspace with user configurable tuning sets
+func CreatePods(f *framework.Framework, name, namespace string, labels labels.Set, spec v1.PodSpec, maxCount int, tuning *TuningSet) error {
+	for i := 0; i < maxCount; i++ {
+		framework.Logf("%v/%v : Creating pod", i+1, maxCount)
+		podObj := newPod(name, namespace, i, labels, spec)
+		if _, err := createNewPodWithRetries(f, namespace, podObj); err != nil {
+			return err
+		}
+		if tuning == nil {
+			continue
+		}
+		// If a rate limit has been defined we wait for N ms between creation
+		if err := tuning.Pods.Sleep(); err != nil {
+			framework.Failf("Unable to get tuning: %v", err)
```

Comment:
I would like to avoid "framework.Failf" in all helper functions. Instead just return an error here.